typescript/common-resolvers/src/label-class.ts

```diff
@@ -106,6 +99,48 @@ const createLabelClass = async (
   )({ id: newLabelClassEntity.id }, user);
 };
 
+const createManyLabelClasses = async (
+  _: any,
+  args: MutationCreateManyLabelClassesArgs,
+  { repository, user }: Context
+): Promise<DbLabelClass[]> => {
+  const { labelClasses: newLabelClasses, datasetId } = args.data;
+  newLabelClasses.forEach((item) => {
+    if (item.datasetId !== datasetId)
+      throw new Error(
+        "Attempted to create many label classes with mismatching dataset IDs"
+      );
+  });
+  const existingLabelClasses = await repository.labelClass.list({
+    datasetId,
+    user,
+  });
+  const nowIso = new Date().toISOString();
+  const newColors = getManyNextClassColors(
+    existingLabelClasses.map((item) => item.color),
+    newLabelClasses.length
+  );
+  const newLabelClassesEntities = newLabelClasses.map(
+    (newLabelClass, index) => ({
+      id: newLabelClass.id ?? uuidv4(),
```

Comment:
But we don't need optimistic responses for this case. We should wait for the creation to complete and show an error in the modal. As for the workspace and dataset upserts.

Reply:
My bad I already removed the uuid that used to be generated before the GQL query, this one is for another reason, and is done like that everywhere else in the common resolvers: it's to be able to return the re-query the DB using those IDs to return the inserted items data.
But yeah it's not the right way, at first I thought it was a "lazy" mistake from the person that coded our "addMany" repository function, since Prisma should return those IDs in the first place.

But no ... https://github.com/prisma/prisma/issues/8131
Also relevant ... https://www.reddit.com/r/node/comments/mip2lz/how_the_f_does_anyone_use_prisma_in_production/